pkg/docker/docker.go

```diff
@@ -900,8 +894,21 @@ func (d *stiDocker) holdHijackedConnection(tty bool, inputStream io.Reader, outp
 }
 
 // RunContainer creates and starts a container using the image specified in opts
-// with the ability to stream input and/or output.
+// with the ability to stream input and/or output.  Any non-nil
+// opts.Std{in,out,err} will be closed upon return.
 func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
+	// Ensure that Std{in,out,err} are closed upon return, including under error
+	// circumstances.
```

Comment:
i thought we wanted to leave stderr open in some cases so the caller than read the container's stderr, and only close it if we failed before starting the container?

(as in what sti.go:685 is doing)